spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkStagedScan.java

```diff
@@ -39,9 +40,8 @@ class SparkStagedScan extends SparkScan {
 
   private List<ScanTaskGroup<ScanTask>> taskGroups = null; // lazy cache of tasks
 
-  SparkStagedScan(SparkSession spark, Table table, SparkReadConf readConf) {
-    super(spark, table, readConf, table.schema(), ImmutableList.of(), null);
-
+  SparkStagedScan(SparkSession spark, Table table, Schema expectedSchema, SparkReadConf readConf) {
+    super(spark, table, readConf, expectedSchema, ImmutableList.of(), null);
```

Comment:
I think you will need to add `readSchema()` to `equals` and `hashCode` now, like we do in other scans that support column pruning. See `SparkBatchQueryScan`.